src/test/java/org/fogbowcloud/manager/core/ApplicationFacadeTest.java

```diff
@@ -12,1701 +12,2209 @@
 
-		Order order = createComputeOrder();
-		OrderStateTransitioner.activateOrder(order);
+    // test case: When calling the method deleteCompute(), the Order passed per parameter must
+    // return its state to Closed.
+    @Test
+    public void testDeleteComputeOrder() throws Exception {
 
-		Mockito.doReturn(order.getFederationUser()).when(this.aaaController).getFederationUser(Mockito.anyString());
+        // set up
+        Order order = createComputeOrder();
+        OrderStateTransitioner.activateOrder(order);
 
-		Mockito.doNothing().when(this.aaaController).authorize(Mockito.any(FederationUser.class),
-				Mockito.any(Operation.class), Mockito.any(Order.class));
+        Mockito.doReturn(order.getFederationUser()).when(this.aaaController)
+                .getFederationUser(Mockito.anyString());
 
-		String federationTokenValue = "";
-		this.application.deleteCompute(order.getId(), federationTokenValue);
+        Mockito.doNothing().when(this.aaaController).authenticate(Mockito.anyString());
 
-		Assert.assertEquals(OrderState.CLOSED, order.getOrderState());
-	}
+        Mockito.doNothing().when(this.aaaController).authorize(Mockito.any(FederationUser.class),
+                Mockito.any(Operation.class), Mockito.any(Order.class));
 
-	@Test
-	public void testDeleteComputeOrderUnathenticated() throws Exception {
-		Mockito.doThrow(new UnauthenticatedUserException()).when(this.aaaController).authenticate(Mockito.anyString());
+        // exercise
+        this.application.deleteCompute(order.getId(), FEDERATION_TOKEN_VALUE);
 
-		Order order = createComputeOrder();
-		OrderStateTransitioner.activateOrder(order);
+        // verify
+        Assert.assertEquals(OrderState.CLOSED, order.getOrderState());
+    }
+
+    // test case: When try calling the deleteCompute() method without user authentication, it must
+    // throw UnauthenticatedUserException and the Order remains in the same state.
+    @Test
+    public void testDeleteComputeOrderUnathenticated() throws Exception {
+
+        // set up
+        Order order = createComputeOrder();
+        OrderStateTransitioner.activateOrder(order);
 
-		Mockito.doReturn(order).when(this.orderController).getOrder(Mockito.anyString(),
-				Mockito.any(FederationUser.class), Mockito.any(InstanceType.class));
+        Mockito.doReturn(order).when(this.orderController).getOrder(Mockito.anyString(),
+                Mockito.any(FederationUser.class), Mockito.any(InstanceType.class));
 
-		Mockito.doNothing().when(this.aaaController).authorize(Mockito.any(FederationUser.class),
-				Mockito.any(Operation.class), Mockito.any(Order.class));
+        Mockito.doNothing().when(this.aaaController).authorize(Mockito.any(FederationUser.class),
+                Mockito.any(Operation.class), Mockito.any(Order.class));
+
+        Mockito.doThrow(new UnauthenticatedUserException()).when(this.aaaController)
+                .authenticate(Mockito.anyString());
```

Comment:
why mocking this, if we do not call authenticate in the method under test?

Reply:
The 'deleteCompute()' method calls 'deleteOrder()' from the 'ApplicationFacade' class. This method 'deleteOrder()' before delegating its effective action to remove the order, requires the AaController class to authenticate the 'federationTokenValue'. And apparently 'Mock' is used to catch the exception thrown in the 'authenticate()' method of this class... 
But for me, this should only be tested in the 'AaControllerTest' class.